### PR TITLE
refactored logging for inclusion on gossip channels.

### PR DESCRIPTION
**Pull request by rolfyone**
investigating #8716 

## Documentation

- [X] I thought about documentation and added the `doc-change-required` label to this PR if updates are required.

## Changelog

- [X] I thought about adding a changelog entry, and added one if I deemed necessary.

### CODE REVIEW

File: networking/p2p/src/main/java/tech/pegasys/teku/networking/p2p/libp2p/gossip/GossipHandler.java [hunk 1]
```diff
@@ -87,6 +88,10 @@ public void gossip(final Bytes bytes) {
     SafeFuture.of(publisher.publish(Unpooled.wrappedBuffer(bytes.toArrayUnsafe()), topic))
         .finish(
             () -> LOG.trace("Successfully gossiped message on {}", topic),
-            err -> LOG.debug("Failed to gossip message on " + topic, err));
```
Comment (rolfyone) [inline networking/p2p/src/main/java/tech/pegasys/teku/networking/p2p/libp2p/gossip/GossipHandler.java:90]: the full stack here got pretty unusable pretty quickly.

File: networking/eth2/src/test/java/tech/pegasys/teku/networking/eth2/gossip/GossipFailureLoggerTest.java [hunk 2]
```diff
@@ -43,7 +43,7 @@ void shouldLogAlreadySeenErrorsAtDebugLevel() {
   void shouldLogFirstNoPeersErrorsAtWarningLevel() {
     try (final LogCaptor logCaptor = LogCaptor.forClass(GossipFailureLogger.class)) {
       logger.logWithSuppression(
-          new RuntimeException("Foo", new NoPeersForOutboundMessageException("So Lonely")), SLOT);
+          new RuntimeException("Foo", new SemiDuplexNoOutboundStreamException("So Lonely")), SLOT);
```
Comment (rolfyone) [inline networking/eth2/src/test/java/tech/pegasys/teku/networking/eth2/gossip/GossipFailureLoggerTest.java]: this SemiDuplex exception used to share implementation with NoPeers error handling so this still tests the same thing and all the logic continues to work.

File: networking/eth2/src/main/java/tech/pegasys/teku/networking/eth2/gossip/GossipFailureLogger.java [hunk 3]
```diff
@@ -47,8 +47,14 @@ public synchronized void logWithSuppression(final Throwable error, final UInt64
...
+      LOG.log(
+          suppress ? Level.DEBUG : Level.WARN,
+          "Failed to publish {}(s) for slot {}; {}",
+          messageType,
+          lastErroredSlot,
+          rootCause.getMessage());
+    } else if (lastRootCause instanceof SemiDuplexNoOutboundStreamException) {
       LOG.log(
           suppress ? Level.DEBUG : Level.WARN,
           "Failed to publish {}(s) for slot {} because no peers were available on the required gossip topic",
```
Comment (tbenr) [inline networking/eth2/src/main/java/tech/pegasys/teku/networking/eth2/gossip/GossipFailureLogger.java]: this seems to me the message for `NoPeersForOutboundMessageException`. And the above seems a more generic one so matching `SemiDuplexNoOutboundStreamException`

**Review by tbenr — APPROVED** [review]
LGTM

**Pull request merged**